Frontend/ui-customer/src/Screens/ZooTicketBookingFlow/TicketBookingScreen/Controller.purs

```diff
@@ -51,44 +59,66 @@ data Action = AfterRender
+  if peopleCat.currentValue == limit then do
+    let updatedData = markLimitCrossed peopleCat true state.data.servicesInfo
+    continue state { data {servicesInfo = markLimitCrossed peopleCat true state.data.servicesInfo}}
+  else do
+    let {finalAmount, updateServicesInfo} = calculateTicketAmount peopleCat (Just true) state state.data.servicesInfo
+    continue state { data { totalAmount = finalAmount, servicesInfo = updateServicesInfo } }
 
-eval (DecrementTicket ticketOption) state = do
-  let {finalAmount, updateServicesInfo} = calculateTicketAmount ticketOption false state.data.servicesInfo
-  continue state { data { totalAmount = finalAmount, servicesInfo = updateServicesInfo } }
+eval (DecrementTicket peopleCat limit) state = do
+  let {finalAmount, updateServicesInfo} = calculateTicketAmount peopleCat (Just false) state state.data.servicesInfo
+      limitUpdatedServicesInfo = markLimitCrossed peopleCat false updateServicesInfo
+  continue state { data { totalAmount = finalAmount, servicesInfo = limitUpdatedServicesInfo } }
 
 eval  (DatePicker _ resp year month date ) state = do
   case resp of 
     "SELECTED" -> do 
-      let validDate = (unsafePerformEffect $ runEffectFn2 compareDate ((show date) <> "/" <> (show (month+1)) <> "/" <> (show year)) (getCurrentDate "" ))
-      continue state {props{validDate = validDate },data { dateOfVisit = (show year) <> "-" <> (show (month+1)) <> "-" <> (show date) }}
+      let selectedDateString = (show year) <> "-" <> (if (month + 1 < 10) then "0" else "") <> (show (month+1)) <> "-" <> (if date < 10 then "0"  else "") <> (show date)
+          validDate = (unsafePerformEffect $ runEffectFn2 compareDate (getDateAfterNDaysv2 (getLimitOfDaysAccToPlaceType state)) selectedDateString)
+                      && (unsafePerformEffect $ runEffectFn2 compareDate selectedDateString (getCurrentDatev2 "" ))
+          modifiedServicesData = map (modifyServicSelectedBHid selectedOpDay selectedDateString) state.data.servicesInfo
+          selectedOpDay = convertUTCtoISC ((show year) <> "-" <> (if (month + 1 < 10) then "0" else "") <> (show (month + 1)) <> "-" <> (show date)) "dddFull"
+      continue state { props{selectedOperationalDay = selectedOpDay, validDate = validDate },data { totalAmount = 0, servicesInfo = modifiedServicesData, dateOfVisit = selectedDateString }}
     _ -> continue state
+  where
+    modifyServicSelectedBHid selectedOpDay selectedDateString service = service {businessHours = map initBusinessHours service.businessHours, selectedSlot = Nothing, selectedBHid = getValidBHid (getValidTimeIntervals service.timeIntervalData selectedOpDay) selectedOpDay selectedDateString state service.expiry}
+    initBusinessHours bh = bh { categories = map (initCategories bh) bh.categories }
+    initCategories bh cat = cat {isSelected = if length bh.categories > 1 then false else true, peopleCategories = map initPeopleCategories cat.peopleCategories}
```

Comment:
refactor this `if length bh.categories > 1 then false else true`

Reply:
done